lightly/api/api_workflow_client.py

```diff
@@ -174,15 +174,19 @@ def upload_file_with_signed_url(self,
             signed_write_url:
                 The url to upload the file to. As no authorization is used,
                 the url must be a signed write url.
+            headers:
+                Specific headers for the request.
 
         Returns:
             The response of the put request, usually a 200 for the success case.
 
         """
+        if headers is not None:
+            response = requests.put(signed_write_url, data=file, headers=headers)
+        else:
+            response = requests.put(signed_write_url, data=file)
 
-        response = requests.put(signed_write_url, data=file)
-
-        if response.status_code != 200:
+        if response.status_code not in [200, 201]:
```

Comment:
nice one!

Reply:
possibly we could make this a response.status_code>=200 and response.status_code<300